File: OpenCVによる画像処理入門/第3版/第2刷/sampleprogram/11-3-5.cpp
#define _USE_MATH_DEFINES
#include <iostream>
#include <cmath>
#include <opencv2/opencv.hpp>
using namespace std;
using namespace cv;
string win_src = "src";
string win_dst = "dst";

int main()
{
  string file_src = "src_depth_hand.png"; // 入力画像のファイル名
  string file_dst = "dst.png"; // 出力画像のファイル名
  Mat img_src = imread(file_src, IMREAD_GRAYSCALE); // 入力画像（グレースケール）の読み込み

  Mat img_dst;
  if(!img_src.data) {
    cout << "error" << endl;
    return -1;
  }

  // ウインドウ生成
  namedWindow(win_src, WINDOW_AUTOSIZE);
  namedWindow(win_dst, WINDOW_AUTOSIZE);

  int depth_min = 173;
  int depth_max = 254;
  inRange(img_src, depth_min, depth_max, img_dst);

  imshow(win_src, img_src);
  imshow(win_dst, img_dst);

  imwrite(file_dst, img_dst);
  waitKey(0);

  return 0;
}

File: OpenCVによる画像処理入門/第3版/第2刷/sampleprogram/11-3-5.py
import cv2
import math
import numpy as np

file_src = 'src_depth_hand.png'
file_dst = 'dst.png'

img_src = cv2.imread(file_src, cv2.IMREAD_GRAYSCALE)

cv2.namedWindow('src')
cv2.namedWindow('dst')

depth_min = 173
depth_max = 254
img_dst = cv2.inRange(img_src, depth_min, depth_max)

cv2.imshow('src', img_src)
cv2.imshow('dst', img_dst)
cv2.imwrite(file_dst, img_dst)
cv2.waitKey(0)
cv2.destroyAllWindows()
